mayavi/tests/test_garbage_collection.py

```diff
@@ -0,0 +1,74 @@
+""" Tests for the garbage collection of objects in mayavi package.
+"""
+# Authors: Deepak Surti, Ioannis Tziakos
+# Copyright (c) 2015, Enthought, Inc.
+# License: BSD Style.
+
+from tvtk.tests.common import TestGarbageCollection
+from mayavi.core.engine import Engine
+from mayavi.core.null_engine import NullEngine
+from mayavi.core.ui.mayavi_scene import MayaviScene
+from mayavi.tools.mlab_scene_model import MlabSceneModel
+
+class TestMayaviGarbageCollection(TestGarbageCollection):
+    """ See: tvtk.tests.common.TestGarbageCollection
+    """
+    def test_mlab_scene_model(self):
+        """ Tests if MlabSceneModel can be garbage collected."""
+        def create_fn():
+            return MlabSceneModel()
```

Comment:
I think that we need both tests types but the gui test is probably more important. One can use the `invoke_after` method of the GUI class to call a function that will close the window.